action/recipes/recipes.go

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+
+// Package recipes yay!
+package recipes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"dagger.io/dagger"
+	"github.com/9elements/firmware-action/action/container"
+	"github.com/sethvargo/go-githubactions"
+)
+
+//===================
+// Universal options
+//===================
+
+var errRequiredOptionUndefined = errors.New("required option is undefined")
+
+type (
+	getValFunc func(string) string
```

Comment:
Currently the linter is rather lax, otherwise it'll complain here, that this line is not formatted right:
`type getValFunc func(string) string` as you'd normally only use braces when there are multiple type declarations back to back. (Just as in C less characters are preferred)